### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by olejandro**

### CODE REVIEW

**Review by siddharth-krishna — APPROVED** [review]

**Comment by olejandro** [comment]
@siddharth-krishna the regression on Ireland seems to be due to adding `re.IGNORECASE` on line 198 in `create_regexp` in `utils`. Somehow, when used in `df.filter`, the anchors for start/end of string are now ignored and substrings are matched. Do you know what's wrong?

**Comment by olejandro** [comment]
@siddharth-krishna I finally got this one. It was simple, but took some time... Could you give another look whether the changes I've done look sensible?

File: xl2times/utils.py [hunk 1]
```diff
@@ -186,22 +186,17 @@ def create_regexp(pattern):
         pattern = remove_negative_patterns(pattern)
     if len(pattern) == 0:
         return re.compile(pattern)  # matches everything
-    # escape special characters
-    # Backslash must come first
-    special = "\\.|^$+()[]{}"
-    for c in special:
-        pattern = pattern.replace(c, "\\" + c)
```
Comment (siddharth-krishna) [inline xl2times/utils.py:193]: We don't need this any more?

File: xl2times/transforms.py [hunk 2]
```diff
@@ -1846,70 +1846,67 @@ def get_matching_commodities(row, dictionary):
...
 def generate_topology_dictionary(
     tables: Dict[str, DataFrame], model: datatypes.TimesModel
 ) -> Dict[str, DataFrame]:
     # We need to be able to fetch processes based on any combination of name, description, set, comm-in, or comm-out
     # So we construct tables whose indices are names, etc. and use pd.filter
 
     dictionary = dict()
+    pros = model.processes
+    coms = model.commodities
+    pros_and_coms = tables[datatypes.Tag.fi_t]
```
Comment (siddharth-krishna) [inline xl2times/transforms.py:1875]: Haha, nice name :)

**Review by siddharth-krishna — APPROVED** [review]

File: xl2times/utils.py (hunk 1 above)
Comment (olejandro) [reply xl2times/utils.py:193]: I believe we don't. Removing it didn't cause any regressions or errors...

Comment (siddharth-krishna) [reply xl2times/utils.py:193]: I'm wondering if we should add a warning/error if we find any special characters in the pattern? If someone uses a model that uses a special character, the failure might be silent / hard to debug.

**Pull request merged**